sourcegraph/sourcegraph
# (analytics) add repo meta usage ping

Part of https://github.com/sourcegraph/pr-faqs/issues/96. Related to https://github.com/sourcegraph/analytics/pull/928.

This PR
- adds backend event logs on repo metadata create/update/delete
- adds repo metadata daily/weekly/monthly usage ping

<details>
<summary>Example:</summary>

```json
"repoMetadataUsage": {
        "Daily": {
            "StartTime": "2023-05-14T00:00:00Z",
            "SearchFilterUsage": {
                "UserCount": 1,
                "EventsCount": 2
            },
            "CreateRepoMetadata": {
                "UserCount": 1,
                "EventsCount": 4
            },
            "DeleteRepoMetadata": {
                "UserCount": 1,
                "EventsCount": 1
            },
            "UpdateRepoMetadata": {
                "UserCount": 1,
                "EventsCount": 1
            }
        },
        "Weekly": {
            "StartTime": "2023-05-14T00:00:00Z",
            "SearchFilterUsage": {
                "UserCount": 1,
                "EventsCount": 2
            },
            "CreateRepoMetadata": {
                "UserCount": 1,
                "EventsCount": 4
            },
            "DeleteRepoMetadata": {
                "UserCount": 1,
                "EventsCount": 1
            },
            "UpdateRepoMetadata": {
                "UserCount": 1,
                "EventsCount": 1
            }
        },
        "Monthly": {
            "StartTime": "2023-05-01T00:00:00Z",
            "SearchFilterUsage": {
                "UserCount": 2,
                "EventsCount": 25
            },
            "CreateRepoMetadata": {
                "UserCount": 1,
                "EventsCount": 4
            },
            "DeleteRepoMetadata": {
                "UserCount": 1,
                "EventsCount": 1
            },
            "UpdateRepoMetadata": {
                "UserCount": 1,
                "EventsCount": 1
            }
        },
        "IsEnabled": false
    },
```
</details>

## Test plan
- Check the diff
- `sg start`
- Open https://sourcegraph.test:3443/site-admin/pings
- Search for `repoMetadataUsage` field and check its structure
